### PR TITLE
Fix/helper text ids

### PR DESCRIPTION
**Pull request by marvinLaubenstein**
### **Preventing/Solving problems with helper-text id**

**Affected components:**

- text-field
- dropdown
- dropdown-select
- radiobutton
- radiobutton-group
- checkbox
- input
- textarea
- date-picker

### CODE REVIEW

**Comment by marvinLaubenstein** [comment]
Hi @acstll, it would be great if you could look over it, since you already know the topic and know what it's about. 

Thank you 🙂

**Comment by nowseemee** [comment]
@acstll can you review this, please?

**Comment by acstll** [comment]
Will do!

File: packages/components/src/components/checkbox/checkbox.tsx [hunk 1]
```diff
@@ -150,7 +150,7 @@ export class Checkbox {
       return (
         <div
           part="helper-text"
-          id={text.id}
+          id={text.internalId}
```
Comment (acstll) [inline packages/components/src/components/checkbox/checkbox.tsx:153]: I think this is a bug, `text.id` seems correct. (Honestly, I would avoid passing anything to the `renderHelperText` method and get the data needed directly in the method itself.)

**Review by acstll — CHANGES_REQUESTED** [review]
I think this is much better than before, but still I guess there's room for duplicated ids within an app (e.g. there could be 2 simultaneus `helper-text-1`). Do you think we can use an util function like [this one in `shoelace`](https://github.com/shoelace-style/shoelace/blob/next/src/internal/auto-increment.ts)?

File: packages/components/src/components/checkbox/checkbox.tsx (hunk 1 above)
Comment (marvinLaubenstein) [reply packages/components/src/components/checkbox/checkbox.tsx]: Oh yes, that's right, I unfortunately slipped in between

**Comment by marvinLaubenstein** [comment]
@acstll I have changed the way now so that the ids were generated by the utils file. I also checked it and could not find any id duplications. Great idea, thanks 🙂

File: packages/components/src/components/textarea/textarea.tsx [hunk 2]
```diff
@@ -107,10 +105,11 @@ export class Textarea {
   @State() hasFocus: boolean = false;
 
   private focusableElement: HTMLElement;
+  private readonly internalId = generateUniqueId();
 
   componentWillLoad() {
     if (this.inputId == null) {
-      this.inputId = 'input-textarea' + i++;
+      this.inputId = 'input-textarea' + this.internalId;
```
Comment (acstll) [inline packages/components/src/components/textarea/textarea.tsx:112]: I would add a dash after "textarea" for consistency (thanks for adding it in the other components) — the rest is ace 👌

**Review by acstll — APPROVED** [review]

File: packages/components/src/components/textarea/textarea.tsx (hunk 2 above)
Comment (marvinLaubenstein) [reply packages/components/src/components/textarea/textarea.tsx:112]: Oh yes, thank you 🙂

**Pull request merged**